### PR TITLE
Feature: implement necessary tweaks to make the Webhooks addon properly work

### PR DESCRIPTION
**Pull request by glaubersilva**
## Description

This PR implements the necessary tweaks to make some features implemented on the new GiveWP Webhooks addon work as expected. The most important change on PR is the Action Scheduler package added as a dependency from Give core, this is a vital package to make the new Webhooks addon work.

This PR also changes how the profile update made using the Donor Dashboard works. Instead of updating info (first name, last name, and email) once per time, we use models to update all at once. This way we can prevent that multiple Webhooks Events get triggered because of just one update.

To make the `subscription_updated` and `subscription_status_changed` events work through the legacy code, was necessary to add to the codebase the following hooks:

```
give_recurring_pre_update_subscription
give_recurring_update_subscription_status
```

## Affects

The new GiveWP Webhooks addon.

## Testing Instructions

1. Pull the changes from this PR
2. Pull the changes from the Webhooks addon, most precisely this branch: https://github.com/impress-org/give-webhooks/pull/7
3. Run the unit tests and make sure everything works
4. To test the donor dashboard update, just create a new webhook with the `donor_updated` event, update the first name, last name, and email using the donor dashboard, then go to `WP > Tools > Scheduled Actions` page and make sure that just one `donor_updated` event was created

## Pre-review Checklist

-   [ ] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -96,6 +97,11 @@
                 "symfony/polyfill-ctype": {},
                 "symfony/polyfill-mbstring": {}
             }
+        },
+        "installer-paths": {
+            "vendor/{$vendor}/{$name}/": [
+                "woocommerce/action-scheduler"
+            ]
         }
```
[the remark this answers is not in the document]
Comment (glaubersilva) [reply composer.json:105]: @JasonTheAdams Without that, the action scheduler is installed in the following path: 
`wp-content/plugins/give/wp-content/plugins/action-scheduler`

Seems that this problem just happens when the composer file also has the `composer/installers` package.

So, these additional instructions make sure that the action scheduler will be installed in this path:
 `wp-content/plugins/give/vendor/woocommerce/action-scheduler`
 
 More details here: https://sanjeebaryal.com.np/action-scheduler-installed-inside-wp-content-folder/

[the remark this answers is not in the document]
Comment (JasonTheAdams) [reply composer.json:105]: Ohhh, right, because it's a WordPress plugin type composer package. Thanks!

**Review by JasonTheAdams — CHANGES_REQUESTED** [review]
I left some comments, @glaubersilva. There's one piece in particular that I think would be better to boy scout code to avoid future confusion.

File: src/DonorDashboards/Profile.php [hunk 2]
```diff
@@ -33,20 +35,24 @@ public function __construct()
...
+
         $pipeline = (new DonorProfilePipeline)
-            ->pipe(new UpdateDonorName)
+            ->pipe(new UpdateDonorName($donor))
             ->pipe(new UpdateDonorCompany)
             ->pipe(new UpdateDonorAvatar)
-            ->pipe(new UpdateDonorEmails)
+            ->pipe(new UpdateDonorEmails($donor))
             ->pipe(new UpdateDonorAddresses)
             ->pipe(new UpdateDonorAnonymousGiving);
```
[the remark this answers is not in the document]
Comment (glaubersilva) [reply src/DonorDashboards/Profile.php]: My initial idea was to use just the model. But I realized the pipeline is updating info not present in the Donor model properties, like avatar and company name for example. 

This way, to not mix it too much and let the thing less messy, now I removed the name and email from the pipeline and did the updating using just the model. In the pipeline, I left just the info that I didn't find a correspondent property in the model.

Resolved in 4890463f02184f67563f835c5d1a6409afeef0ba

File: src/DonorDashboards/Pipeline/Stages/UpdateDonorName.php [hunk 3]
```diff
@@ -2,50 +2,48 @@
...
+    /**
+     * @var Donor
+     */
     protected $donor;
 
-    public function __invoke($payload)
+    /**
+     * @unreleased
+     */
+    public function __construct(Donor &$donor)
```
[the remark this answers is not in the document]
Comment (glaubersilva) [reply src/DonorDashboards/Pipeline/Stages/UpdateDonorName.php]: Resolved in 4890463f02184f67563f835c5d1a6409afeef0ba

File: src/DonorDashboards/Pipeline/Stages/UpdateDonorName.php [hunk 4]
```diff
@@ -2,50 +2,48 @@
...
-
-        $this->updateNameInMetaDB();
-        $this->updateNameInDonorDB();
-
-        return $payload;
+        $this->donor = &$donor;
     }
 
-    protected function updateNameInMetaDB()
+    /**
```
[the remark this answers is not in the document]
Comment (glaubersilva) [reply src/DonorDashboards/Pipeline/Stages/UpdateDonorName.php]: Resolved in 4890463f02184f67563f835c5d1a6409afeef0ba

**Review by JasonTheAdams — APPROVED** [review]
We decided to clean up the pipeline in a subsequent PR. This is sufficient for now. Good work, @glaubersilva!

**Pull request merged**